substrafl/model_loading.py

```diff
@@ -147,93 +144,191 @@ def _get_train_task_from_round(client: substra.Client, compute_plan_key: str, ro
+
+    return local_task
+
+
+def _load_instance(gz_path: Path, extraction_folder: Path, remote: bool) -> Any:
+    """Load into memory a serialized (and compressed (.tar.gz)) SubstraFL Remote Struct instance within the
+    given path.
     This kind of file is usually the result of the ``substra.Client.download_function`` function applied to
-    a train task being part of a Substrafl experiment.
+    a task being part of a SubstraFL experiment.
 
     Args:
-        algo_path (Path): A file being the tar.gz compression of a substrafl RemoteStruct algorithm
+        gz_path (Path): A file being the tar.gz compression of a SubstraFL RemoteStruct instance
         extraction_folder (Path): Where to unpack the folder.
+        remote (bool): Whether to return the instance or the remote instance from the loaded remote_struct.
 
     Returns:
-        Any: The loaded substrafl object into memory.
+        Any: The loaded SubstraFL object into memory.
     """
-
-    with tarfile.open(algo_path, "r:gz") as tar:
+    with tarfile.open(gz_path, "r:gz") as tar:
         tar.extractall(path=extraction_folder)
 
     sys.path.append(str(extraction_folder))  # for local dependencies
 
     remote_struct = RemoteStruct.load(src=extraction_folder / SUBSTRAFL_FOLDER)
 
-    my_algo = remote_struct.get_instance()
+    if not remote:
+        instance = remote_struct.get_instance()
+    else:
+        instance = remote_struct.get_remote_instance()
+
+    return instance
+
 
-    return my_algo
+def _load_from_files(input_folder: os.PathLike, remote: bool = False) -> Any:
+    """Loads an instance from a specified folder. This folder should contains:
 
+        - function.tar.gz
+        - metadata.json
+        - the file specified in metadata.model_file
 
-def download_algo_files(
+    Args:
+        input_folder (os.PathLike): Path to folder containing the required files.
+        remote (bool): Wether the the instance to load is to load using a local method or a remote method.
+
+    Raises:
+        exceptions.LoadMetadataError: The metadata file must contains the model_file key
+        exceptions.LoadFileNotFoundError: At least one of the required file to load the instance is not found
+        exceptions.LoadLocalDependencyError: One of the dependency used by the instance is not installed within the
+          used environment
+
+    Returns:
+        Any: The serialized instance within the input_folder
+    """
+
+    folder = Path(input_folder)
+
+    metadata = _validate_folder_content(folder=folder)
+
+    _check_environment_compatibility(metadata=metadata)
+
+    try:
+        instance = _load_instance(gz_path=folder / metadata[FUNCTION_DICT_KEY], extraction_folder=folder, remote=remote)
+
+        if not remote:
+            loaded_instance = instance.load_local_state(folder / metadata[MODEL_DICT_KEY])
+        else:
+            loaded_instance = instance.load_model(folder / metadata[MODEL_DICT_KEY])
+
+    except ModuleNotFoundError as e:
```

Comment:
do we except the first or the second half to raise? would it be possible to have a narrower try / except encapsulation?